dace/codegen/targets/fpga.py

```diff
@@ -386,20 +391,28 @@ def generate_scope(self, sdfg, dfg_scope, state_id, function_stream,
 
     def allocate_array(self, sdfg, dfg, state_id, node, function_stream,
                        declaration_stream, allocation_stream):
+
         result_decl = StringIO()
         result_alloc = StringIO()
         nodedesc = node.desc(sdfg)
         arrsize = nodedesc.total_size
         is_dynamically_sized = dace.symbolic.issymbolic(arrsize, sdfg.constants)
-
         dataname = node.data
+
+        if not isinstance(nodedesc, dace.data.Stream):
```

Comment:
could you explain?

Reply:
Part of the codegen relies on the fact that allocate_array will return if the array has been already allocated.
This was not done on the FPGA backend, so I added it.
There is this special case for Stream, because we need to keep track of the names of the streams across nested SDFG for Intel FPGA channel mangling (essentially we need to call allocate_stream for this)

I can also leave it like it was before, without returning if the array has been already allocated

Reply:
This makes sense, but a comment (with this content) would be welcome and helpful :)

Reply:
Will add it!